homeassistant/components/netatmo/climate.py

```diff
@@ -175,41 +222,72 @@ def __init__(self, data, room_id):
-        self.update_without_throttle = False
-        self._module_type = self._data.room_status.get(room_id, {}).get(
-            "module_type", NA_VALVE
-        )
-
-        if self._module_type == NA_THERM:
-            self._operation_list.append(HVAC_MODE_OFF)
 
-        self._unique_id = f"{self._room_id}-{self._module_type}"
+        self._away_temperature = None
+        self._hg_temperature = None
+        self._boilerstatus = None
+        self._setpoint_duration = None
 
-    @property
-    def device_info(self):
-        """Return the device info for the thermostat/valve."""
-        return {
-            "identifiers": {(DOMAIN, self._room_id)},
-            "name": self._room_name,
-            "manufacturer": MANUFACTURER,
-            "model": MODELS[self._module_type],
-        }
+        if self._model == NA_THERM:
+            self._operation_list.append(HVAC_MODE_OFF)
 
-    @property
-    def unique_id(self):
-        """Return a unique ID."""
-        return self._unique_id
+        self._unique_id = f"{self._id}-{self._model}"
+
+    async def async_added_to_hass(self) -> None:
+        """Entity created."""
+        await NetatmoBase.async_added_to_hass(self)
+
+        async def handle_event(event):
+            """Handle webhook events."""
+            data = event.data["data"]
+
+            if not data.get("event_type"):
+                return
+
+            if not data.get("home"):
+                return
+
+            home = data["home"]
+            if self._home_id == home["id"] and data["event_type"] == "therm_mode":
+                self._preset = NETATMO_MAP_PRESET[home["therm_mode"]]
+                self._hvac_mode = HVAC_MAP_NETATMO[self._preset]
+                if self._preset == PRESET_FROST_GUARD:
+                    self._target_temperature = self._hg_temperature
+                elif self._preset == PRESET_AWAY:
+                    self._target_temperature = self._away_temperature
+                elif self._preset == PRESET_SCHEDULE:
+                    self.async_update_callback()
+                self.schedule_update_ha_state()
+                return
+
+            if not home.get("rooms"):
+                return
+
+            for room in home["rooms"]:
+                if data["event_type"] == "set_point":
+                    if self._id == room["id"]:
+                        if room["therm_setpoint_mode"] == "off":
+                            self._hvac_mode = HVAC_MODE_OFF
+                        else:
+                            self._target_temperature = room[
+                                "therm_setpoint_temperature"
+                            ]
+                        self.schedule_update_ha_state()
+                        break
+
+                elif data["event_type"] == "cancel_set_point":
+                    if self._id == room["id"]:
+                        self.async_update_callback()
+                        self.schedule_update_ha_state()
```

Comment:
```suggestion
                self.async_write_ha_state()
```
```suggestion
                        self.async_write_ha_state()
```
Can we write state directly instead since we are already in async context?